manim/_config/utils.py

```diff
@@ -660,28 +668,23 @@ def digest_args(self, args: argparse.Namespace) -> "ManimConfig":
 
         # Handle the -n flag.
         nflag = args.from_animation_number
-        if nflag is not None:
-            if "," in nflag:
-                start, end = nflag.split(",")
-                self.from_animation_number = int(start)
-                self.upto_animation_number = int(end)
-            else:
-                self.from_animation_number = int(nflag)
+        if nflag:
+            self.from_animation_number = nflag[0]
+            try:
+                self.upto_animation_number = nflag[1]
+            except:
+                logging.getLogger("manim").info(
```

Comment:
don't do this. Use the logger from rich, usually defined in the root of the module.

Reply:
I just borrowed the code from this [same file here](https://github.com/ManimCommunity/manim/blob/1b344260e1bb8c9c747b7689d49d753d06b8552a/manim/_config/utils.py#L137-L139).

I also think `logging.getLogger("manim")` is getting the [logger from rich](https://github.com/ManimCommunity/manim/blob/1b344260e1bb8c9c747b7689d49d753d06b8552a/manim/_config/logger_utils.py#L85-L90)?

Reply:
you should essentially import `logger` from root of the package
https://github.com/ManimCommunity/manim/blob/1b344260e1bb8c9c747b7689d49d753d06b8552a/manim/_config/utils.py#L25
add a logger import there and do this change
```suggestion
                logger.info(
```

Reply:
Alright, I'll change it on the other place I mentioned as well.

Reply:
After trying to import logger from root of the package, I run into errors regarding circular imports:
```
ImportError: cannot import name 'logger' from partially initialized module 'manim' (most likely due to a circular import) (C:\Users\a\Documents\repos\ManimCommunity\manim\manim\__init__.py)
```
I'm not sure how to deal with this particular issue of circular import given how we're using the logging package, and I'm guessing that's why @leotrs wrote it this way. I move not to change this here since this seems less related to the Argparse -> Click refactor and more so a ManimConfig/_config refactor.